test/Dapr.Client.Test/InvokeApiTest.cs

```diff
@@ -413,6 +423,271 @@ public async Task InvokeMethodAsync_WithReturnTypeAndData_WithNonDefaultVerb_Wit
             response.Name.Should().Be(invokedResponse.Name);
         }
 
+        [Fact]
+        public async Task InvokeMethodAsync_CanInvokeMethodWithResponseHeaders_CalleeSideGrpc()
+        {
+            var client = new MockClient();
+            var data = new Response() { Name = "Look, I was invoked!" };
+            var invokeResponse = new InvokeResponse();
+            invokeResponse.Data = TypeConverters.ToAny(data);
+            
+            var response = 
+                client.Call<InvokeResponse>()
+                .SetResponse(invokeResponse)
+                .AddHeader("some-header", "some-value")
+                .Build();
+
+            
+            client.Mock
+            .Setup(m => m.InvokeServiceAsync(It.IsAny<Autogen.Grpc.v1.InvokeServiceRequest>(), It.IsAny<CallOptions>()))
+            .Returns(response);
+
+            var body = new Request() { RequestParameter = "Hello " };
+            var task = client.DaprClient.InvokeMethodWithResponseHeadersAsync<Request, Response>("test", "testMethod", body);
+
+            // Validate Response
+            var invokedResponse = await task;
+            invokedResponse.Body.Name.Should().Be("Look, I was invoked!");
+            invokedResponse.HttpStatusCode.Should().BeNull();
+            invokedResponse.GrpcStatusInfo.Should().NotBeNull();
+            invokedResponse.GrpcStatusInfo.GrpcStatusCode.Should().Be(Grpc.Core.StatusCode.OK);
+        }
+
+        [Fact]
+        public async Task InvokeMethodAsync_CanInvokeMethodWithResponseHeaders_CalleeSideHttp()
+        {
+            var client = new MockClient();
+            var data = new Response() { Name = "Look, I was invoked!" };
+            var invokeResponse = new InvokeResponse();
+            invokeResponse.Data = TypeConverters.ToAny(data);
+            
+            var response = 
+                client.Call<InvokeResponse>()
+                .SetResponse(invokeResponse)
+                .AddHeader("dapr-http-status", "200")
+                .Build();
+
+            
+            client.Mock
+            .Setup(m => m.InvokeServiceAsync(It.IsAny<Autogen.Grpc.v1.InvokeServiceRequest>(), It.IsAny<CallOptions>()))
+            .Returns(response);
+
+            var body = new Request() { RequestParameter = "Hello " };
+            var task = client.DaprClient.InvokeMethodWithResponseHeadersAsync<Request, Response>("test", "testMethod", body);
+
+            // Validate Response
+            var invokedResponse = await task;
+            invokedResponse.Body.Name.Should().Be("Look, I was invoked!");
+            invokedResponse.GrpcStatusInfo.Should().BeNull();
+            invokedResponse.HttpStatusCode.Should().NotBeNull();
+            invokedResponse.HttpStatusCode.Should().Be(HttpStatusCode.OK);
+        }
+
+        [Fact]
+        public async Task InvokeMethodAsync_CanInvokeMethodWithResponseHeaders_ServerThrowsRpcException()
```

Comment:
`ServerThrowsRpcException`- we should pick a better name here. Servers don't throw exceptions, they return responses.